panther_analysis_tool/schemas.py

```diff
@@ -14,7 +14,9 @@
 limitations under the License.
 '''
 
-from schema import Optional, Or, Schema
+from schema import And, Optional, Or, Regex, Schema
+
+NAME_ID_VALIDATION_REGEX = Regex(r"^[A-Za-z0-9\-\_\.\ \(\)]+$")
```

Comment:
Nit: inside a character group you typically don't need escaping. This should be equivalent:

```suggestion
NAME_ID_VALIDATION_REGEX = Regex(r"^[A-Za-z0-9_. ()-]+$")
```

A little easier to read without all the escapes. (Note the `-` is now at the end so it will be interpreted literally.)